Makefile

```
@@ -381,8 +381,9 @@ ifdef LLAMA_BLIS
 endif # LLAMA_BLIS
 
 ifdef LLAMA_CUBLAS
-	MK_CPPFLAGS  += -DGGML_USE_CUBLAS -I/usr/local/cuda/include -I/opt/cuda/include -I$(CUDA_PATH)/targets/x86_64-linux/include -I/usr/local/cuda/targets/aarch64-linux/include
-	MK_LDFLAGS   += -lcuda -lcublas -lculibos -lcudart -lcublasLt -lpthread -ldl -lrt -L/usr/local/cuda/lib64 -L/opt/cuda/lib64 -L$(CUDA_PATH)/targets/x86_64-linux/lib -L/usr/local/cuda/targets/aarch64-linux/lib -L/usr/lib/wsl/lib
+	CUDA_PATH ?= /usr/local/cuda
+	MK_CPPFLAGS  += -DGGML_USE_CUBLAS -I$(CUDA_PATH)/include -I/opt/cuda/include -I$(CUDA_PATH)/targets/$(UNAME_M)-linux/include
+	MK_LDFLAGS   += -lcuda -lcublas -lculibos -lcudart -lcublasLt -lpthread -ldl -lrt -L$(CUDA_PATH)/lib64 -L/opt/cuda/lib64 -L/usr/lib64 -L$(CUDA_PATH)/targets/$(UNAME_M)-linux/lib -L/usr/lib/wsl/lib
```

e.g. FindCUDA.cmake defaults to /opt/cuda and only uses other paths if an environment variable is set, so wouldn't this work just as well?
```suggestion
	CUDA_PATH ?= /opt/cuda
	MK_CPPFLAGS  += -DGGML_USE_CUBLAS -I$(CUDA_PATH)/include -I$(CUDA_PATH)/targets/$(UNAME_M)-linux/include
	MK_LDFLAGS   += -lcuda -lcublas -lculibos -lcudart -lcublasLt -lpthread -ldl -lrt -L$(CUDA_PATH)/lib64 -L/usr/lib64 -L$(CUDA_PATH)/targets/$(UNAME_M)-linux/lib -L/usr/lib/wsl/lib
```

---

`FindCUDA.cmake` also looks in the system path, this would break many systems.

---

We could use pkg-config if CUDA_PATH is not set, but it's more verbose:
```suggestion
	MK_CPPFLAGS  += -DGGML_USE_CUBLAS
	MK_LDFLAGS   += -lculibos -lcublasLt -lpthread -ldl -lrt
ifneq '' '$(CUDA_PATH)'
	MK_CPPFLAGS  += -I$(CUDA_PATH)/include -I$(CUDA_PATH)/targets/$(UNAME_M)-linux/include
	MK_LDFLAGS   += -lcuda -lcublas -lcudart -L$(CUDA_PATH)/lib64 -L/usr/lib64 -L$(CUDA_PATH)/targets/$(UNAME_M)-linux/lib -L/usr/lib/wsl/lib
else
	MK_CPPFLAGS  += $(shell pkg-config --cflags-only-I cuda cublas cudart)
	MK_LDFLAGS   += $(shell pkg-config --libs cuda cublas cudart)
endif
```

Or choose the fallback CUDA_PATH in a smarter way:
```suggestion
ifneq '' '$(wildcard /opt/cuda)'
	CUDA_PATH ?= /opt/cuda
else
	CUDA_PATH ?= /usr/local/cuda
endif
	MK_CPPFLAGS  += -DGGML_USE_CUBLAS -I$(CUDA_PATH)/include -I$(CUDA_PATH)/targets/$(UNAME_M)-linux/include
	MK_LDFLAGS   += -lcuda -lcublas -lculibos -lcudart -lcublasLt -lpthread -ldl -lrt -L$(CUDA_PATH)/lib64 -L/usr/lib64 -L$(CUDA_PATH)/targets/$(UNAME_M)-linux/lib -L/usr/lib/wsl/lib
```

---

I prefer the second :)